packages/material-ui-lab/src/YearPicker/PickersYear.tsx

```diff
@@ -15,90 +26,130 @@ export interface YearProps {
   value: number;
 }
 
-export type PickersYearClassKey = 'root' | 'modeDesktop' | 'yearButton' | 'disabled' | 'selected';
+export type PickersYearClassKey = keyof typeof pickersYearClasses;
 
-export const styles: MuiStyles<PickersYearClassKey> = (theme): StyleRules<PickersYearClassKey> => ({
-  root: {
-    flexBasis: '33.3%',
-    display: 'flex',
-    alignItems: 'center',
-    justifyContent: 'center',
+type StyleProps = YearProps & { wrapperVariant: WrapperVariant };
+
+const useUtilityClasses = (styleProps: StyleProps) => {
+  const { wrapperVariant, disabled, selected, classes } = styleProps;
+
+  const slots = {
+    root: ['root', wrapperVariant && `mode${capitalize(wrapperVariant)}`],
+    yearButton: ['yearButton', disabled && 'disabled', selected && 'selected'],
+  };
+
+  return composeClasses(slots, getPickersYearUtilityClass, classes);
+};
+
+const PickersYearRoot = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiPickersYear',
+    slot: 'Root',
+    overridesResolver: (
+      props: { styleProps: StyleProps },
+      styles: Partial<Record<PickersYearClassKey, object>>,
+    ) => {
+      const { styleProps } = props;
+      return {
+        ...styles.root,
+        ...(styleProps.wrapperVariant &&
+          styles[`mode${capitalize(styleProps.wrapperVariant)}` as PickersYearClassKey]),
+      };
+    },
   },
-  modeDesktop: {
+)(({ styleProps }) => ({
+  flexBasis: '33.3%',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  ...(styleProps?.wrapperVariant === 'desktop' && {
     flexBasis: '25%',
+  }),
+}));
+
+const PickersYearButton = experimentalStyled(
+  'button' as const,
+  {},
+  {
+    name: 'MuiPickersYear',
+    slot: 'YearButton',
+    overridesResolver: () => ({}),
```

Comment:
I think that it will always be interesting to allow nested slots to be targeted from the theme.

```suggestion
    overridesResolver: (props, styles) => styles.yearButton,
```

Reply:
agree.